sw/device/tests/keymgr_key_derivation_test.c

```diff
@@ -61,6 +61,32 @@ bool test_main(void) {
       keymgr_testutils_generate_versioned_key(&keymgr, sideload_params));
   LOG_INFO("Keymgr generated HW output for Otbn at OwnerIntKey State");
 
+  CHECK_STATUS_OK(
+      keymgr_testutils_advance_state(&keymgr, &kOwnerRootKeyParams));
+  CHECK_STATUS_OK(
+      keymgr_testutils_check_state(&keymgr, kDifKeymgrStateOwnerRootKey));
+  LOG_INFO("Keymgr entered OwnerRootKey State");
+
+  CHECK_STATUS_OK(
+      keymgr_testutils_generate_versioned_key(&keymgr, kKeyVersionedParams));
+  LOG_INFO("Keymgr generated SW output at OwnerRootKey State");
+
+  // Generate sideload keys for 3 HW interfaces - Kmac, Aes, Otbn.
+  sideload_params.dest = kDifKeymgrVersionedKeyDestKmac;
+  CHECK_STATUS_OK(
+      keymgr_testutils_generate_versioned_key(&keymgr, sideload_params));
+  LOG_INFO("Keymgr generated HW output for Kmac at OwnerRootKey State");
+
+  sideload_params.dest = kDifKeymgrVersionedKeyDestAes;
+  CHECK_STATUS_OK(
+      keymgr_testutils_generate_versioned_key(&keymgr, sideload_params));
+  LOG_INFO("Keymgr generated HW output for Aes at OwnerRootKey State");
+
+  sideload_params.dest = kDifKeymgrVersionedKeyDestOtbn;
+  CHECK_STATUS_OK(
+      keymgr_testutils_generate_versioned_key(&keymgr, sideload_params));
+  LOG_INFO("Keymgr generated HW output for Otbn at OwnerRootKey State");
+
```

Comment:
Are these changes necessary on this top-level test? It seems to me like since we are not comparing the key outputs, we are not actually testing the correct generation of values. 

I wonder if its DV component (`chip_sw_keymgr_key_derivation_vseq.sv` and few other sequences derived from this) also needs to be updated so that we can confirm these keys are generated correctly.

Reply:
I suppose it's not really necessary; I just figured it made sense to test the `OwnerRootKey` state as well since it wasn't previously tested. Ideally the DV component would check the expected values, you're right; maybe I should remove it from this change to avoid giving the impression we're testing something we're really not testing.